src/rust/src/lib.rs

```diff
@@ -169,3 +175,1499 @@ extern "C" fn ccxr_close_handle(handle: RawHandle) {
         let _file = File::from_raw_handle(handle);
     }
 }
+
+#[cfg(test)]
+pub mod tests {

```

Comment:
We are testing the parser here, so you can place the tests in `parser.rs`

Reply:
yes, done!